AdminDashboard/BusinessLogicOrchestrators/LoginOrchestrator/LoginOrchestrator.cs

```diff
@@ -0,0 +1,110 @@
+﻿using AdminDashboard.BusinessLogicOrchestrators.LoginOrchestrator;
+using AdminDashboard.Exceptions;
+using AdminDashboard.Main.Configurations;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace GravitationalTest.BusinessOrchestrators.Users
+{
+    public class LoginOrchestrator : ILoginOrchestrator
+    {
+        private readonly string Username = "Minh";
+        private static readonly int SALT_BYTE_SIZE = 16; // byte array size for the salt
+        private static readonly int HASH_BYTE_SIZE = 20; // byte array size for the hash, together with salt is 36 bytes
+
+        private readonly AuthorizationConfiguration Configuration;
+
+        public LoginOrchestrator(IOptions<AuthorizationConfiguration> configuration)
+        {
+            Configuration = configuration.Value;
+        }
+
+        public string Authenticate(string username, string password)
+        {
+            if (UsernameIsIncorrect(username))
+                throw new InvalidCredentialsException();
+
+            if (IsPasswordIncorrect(password))
+                throw new InvalidCredentialsException();
+
+            return GenerateJwtToken();
+        }
+
+        private bool IsPasswordIncorrect(string password)
+        {
+            string savedPasswordHash = Configuration.Secret;
+            byte[] expectedHash = GetBytesFrom(savedPasswordHash);
+            byte[] salt = new byte[SALT_BYTE_SIZE];
+            Array.Copy(expectedHash, 0, salt, 0, SALT_BYTE_SIZE);
+
+            byte[] actualHash = HashPasswordAttempt(password, salt);
+
+            return CompareHashes(expectedHash, actualHash);
+        }
+
+        private static bool CompareHashes(byte[] expectedHash, byte[] actualHash)
```

Comment:
I dislike the name of this function. Just looking at its signature, it's unclear what `CompareHashes` does - I would expect it returns true if the hashes are equal. Maybe something like `IsEqual` would be better - or just inline the for loop in your `IsPasswordIncorrect` method above

Reply:
Yeah, I think I overencapsulated - I put the functionality inline the `IsPasswordIncorrect` function!